src/stdpar/plugin-PixelVertexFinding/gpuSortByPt2.h

```diff
@@ -55,15 +56,15 @@ namespace gpuVertexFinder {
         sortInd[0] = 0;
       return;
     }
-#ifdef __CUDA_ARCH__
-    __shared__ uint16_t sws[1024];
-    // sort using only 16 bits
-    radixSort<float, 2>(ptv2, sortInd, sws, nvFinal);
-#else
-    for (uint16_t i = 0; i < nvFinal; ++i)
-      sortInd[i] = i;
-    std::sort(sortInd, sortInd + nvFinal, [&](auto i, auto j) { return ptv2[i] < ptv2[j]; });
-#endif
+    if target (nv::target::is_device) {
```

Comment:
Just curious, is `if target` `nvc++` extension?

Reply:
yes, this the `nvc++` alternative to `__CUDA_ARCH__`. `nvcc` also support this syntax. It is useful to generate different instructions for a function that is compiled for both device and host contexts or different device architecture. It has some limitation, mainly that it can only be used within function scope and can't leak into type system (e.g. change return type depending of arch).

Reply:
> `nvcc` also support this syntax. 

it does ?

Reply:
this is what Bryce Lelbach said in his [GTC talk](https://www.nvidia.com/en-us/on-demand/session/gtcspring21-s31358/) last year but I did not try it myself so I might be wrong on that